papermill_origami/noteable_dagstermill/assets.py

```diff
@@ -0,0 +1,288 @@
+import copy
+import os
+import sys
+import tempfile
+import uuid
+from base64 import b64encode
+from typing import Any, Dict, Mapping, Optional, Set, Union
+
+import cloudpickle as pickle
+import dagster._check as check
+import nbformat
+import papermill
+from dagster import (
+    AssetIn,
+    AssetKey,
+    MetadataValue,
+    Nothing,
+    Output,
+    PartitionsDefinition,
+    ResourceDefinition,
+    asset,
+)
+from dagster._core.definitions.events import CoercibleToAssetKeyPrefix
+from dagster._core.definitions.utils import validate_tags
+from dagster._core.execution.context.compute import SolidExecutionContext
+from dagster._core.execution.context.system import StepExecutionContext
+from dagster._utils import safe_tempfile_path
+from dagster._utils.error import serializable_error_info_from_exc_info
+from dagstermill import _load_input_parameter, _reconstitute_pipeline_context
+from dagstermill.compat import ExecutionError
+from dagstermill.factory import _find_first_tagged_cell_index, get_papermill_parameters
+from papermill.iorw import load_notebook_node, write_ipynb
+from papermill.translators import PythonTranslator
+
+from ..util import removeprefix
+
+
+def _dm_compute(
+    name,
+    notebook_path,
+    notebook_url,
+):
+    check.str_param(name, "name")
+    check.str_param(notebook_path, "notebook_path")
+    check.str_param(notebook_url, "notebook_url")
+
+    def _t_fn(context, **inputs):
```

Comment:
I think this inner function `_t_fn` can be entirely changed to match the one in `ops.py` and then refactored out into a util to avoid code duplication.

In addition, I think #45 should be merged into this branch so as to consolidate changes in `ops.py` as well.